dvc/repo/reproduce.py

```diff
@@ -35,7 +35,7 @@ def _reproduce_stage(stages, node, force, dry, interactive, no_commit):
 def reproduce(
     self,
     target=None,
-    recursive=True,
+    single_item=False,
```

Comment:
This is breaking the backward compatibility, but since our API is not officially released yet and this change is indeed worthy, I think it is fine.

Reply:
Actually this is just the internal naming, `recursive` argument was originally used to specify the default behaviour (`recursive = not self.args.single_item`), so the behavior stays the same. I just renamed the `recursive` method argument to `single_item` (and removed the negation) and added a new `recursive` argument that corresponds to the `--recursive` option.